### PR TITLE
Add app content

### PR DESCRIPTION
**Pull request by anna-osipova**
A simple Pokedex app with two levels of navigation. I've attempted to use the same libs, lint rules etc. as in previous modules to not create extra confusion.

We could add a search filed to the list page if we want more stuff.

There's jest and eslint, and a few tests. We can break some of those later.

### CODE REVIEW

File: test/App.jest.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  it('shows Loader', async () => {
+    axiosMock.get.mockResolvedValueOnce({})
+    await act(async () => {
+      const { getByAltText } = render(<App />)
+      expect(getByAltText('Loading...')).toBeVisible()
+    })
+  })
+
+  it('shows error', async () => {
+    axiosMock.get.mockResolvedValueOnce(Promise.reject(new Error()))
```
Comment (RikuLa) [inline test/App.jest.spec.jsx:34]: you can also do `.mockRejectedValueOnce`!

File: src/PokemonPage.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  const { type } = pokemon.types.find((type) => type.slot === 1)
+  const stats = pokemon.stats.map((stat) => ({
+    name: stat.stat.name.replace('-', ' '),
+    value: stat.base_stat
+  })).reverse()
+  const normalAbility = pokemon.abilities.find((ability) => !ability.is_hidden)
+  const hiddenAbility = pokemon.abilities.find((ability) => ability.is_hidden === true)
+  const previous = pokemons.find(({ id }) => id === pokemon.id - 1)
+  const next = pokemons.find(({ id }) => id === pokemon.id + 1)
```
Comment (RikuLa) [inline src/PokemonPage.jsx:26]: I feel like these should come in as props. A page for a single Pokemon should really not consume ALL the pokemon.

File: package.json [hunk 3]
```diff
@@ -6,6 +6,7 @@
   "scripts": {
     "start": "webpack-dev-server --open --mode development",
     "test": "jest",
+    "eslint": "eslint './**/*.{js,jsx}'",
```
[the remark this answers is not in the document]
Comment (anna-osipova) [reply package.json:9]: I copied this from the example in some previous module to keep things as familiar as possible for the students.

File: src/styles.css [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+  --pokemon-type-color: #ac2;
+}
+
+.pokemon-type-dark {
+  --pokemon-type-color: #754;
+}
+
+.pokemon-type-dragon {
+  --pokemon-type-color: #73f;
+}
```
[the remark this answers is not in the document]
Comment (anna-osipova) [reply src/styles.css:153]: 😊

File: src/useApi.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from 'react'
+import axios from 'axios'
+
+const useApi = (url, mapResults = (result) => result) => {
+  const [data, setData] = useState()
+  const [isLoading, setIsLoading] = useState(true)
+  const [error, setError] = useState()
+  useEffect(() => {
+    setIsLoading(true)
+    axios
```
[the remark this answers is not in the document]
Comment (anna-osipova) [reply src/useApi.js:10]: They were using `axios` in the React module so I decided to stick with it.

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (RikuLa) [reply package.json:9]: Allright! 👍

File: src/useApi.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (RikuLa) [reply src/useApi.js:10]: Yes, that is then good enough for us as well!

File: public/index.html [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8">
+    <title>Pokemons</title>
```
[the remark this answers is not in the document]
Comment (anna-osipova) [reply public/index.html]: I should've known better! 😱

**Comment by anna-osipova** [comment]
> Really nice app! This was one of my most fun reviews in a long while :D We might want to do some style changes later on ( center the card and loading spinner etc ). I really like the card design in the individual pokemon page!

Glad you like it! I made the card based on some design on the Internet.

I cleaned up the code based on your comments 👍

File: test/PokemonPage.jest.spec.jsx [hunk 7]
```diff
@@ -93,12 +93,12 @@ describe('<PokemonPage />', () => {
...
 
   it('should render stats', async () => {
-    axiosMock.get.mockResolvedValueOnce({ data: pokemon })
+    axiosMock.get.mockResolvedValueOnce({ data: pokemonList })
 
     await act(async () => {
       render(
         <Router history={history}>
-          <PokemonPage pokemons={pokemons} />
+          <PokemonPage pokemonList={pokemons} />
```
Comment (RikuLa) [inline test/PokemonPage.jest.spec.jsx]: The value of the prop is still pokemons 🙈

File: src/App.jsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+import PokemonList from './PokemonList'
+
+const mapResults = (({ results }) => results.map(({ url, name }) => ({
+  url,
+  name,
+  id: parseInt(url.match(/\/(\d+)\//)[1])
+})))
+
+const App = () => {
+  const { data: pokemons, error, isLoading } = useApi('https://pokeapi.co/api/v2/pokemon/?limit=784', mapResults)
```
[the remark this answers is not in the document]
Comment (anna-osipova) [reply src/App.jsx]: Fetch on scroll would make sense, but I don't want to make this too big and complicated.

File: test/PokemonPage.jest.spec.jsx (hunk 7 above)
Comment (anna-osipova) [reply test/PokemonPage.jest.spec.jsx]: Not anymore 🙃

**Pull request merged**